.woodpecker/test.yml

```diff
@@ -1,19 +1,12 @@
-# TODO: The when: platform conditionals aren't working currently
-# See https://github.com/woodpecker-ci/woodpecker/issues/1677
+platform: linux/amd64
+
+when:
+  event: [pull_request, push]
```

Comment:
you need to decide if it should run if merged too ... else just remove push event ...